lua/dap/session.lua

```diff
@@ -112,15 +112,22 @@ local function coresume(co)
 end
 
 
-local function launch_external_terminal(terminal, args)
+local function launch_external_terminal(env, terminal, args)
   local handle
   local pid_or_err
   local full_args = {}
   vim.list_extend(full_args, terminal.args or {})
   vim.list_extend(full_args, args)
+  local env_formatted = {}
+  for k, v in pairs(env or {}) do
+    if k:find "^[^=]*$" then -- correct variable?
+      env_formatted[#env_formatted+1] = k.."="..tostring(v)
+    end
+  end
```

Comment:
I think if `env` is set then `spawn` uses only these variables, without inheriting/merging any from the parent process.
I suspect this would need to include at least `PATH` but might be safer to copy the entire `vim.fn.environ()` and extend it based on `env`

Reply:
Fixed using `vim.extend_tbl` in b8e54ac91cae94df6c609d986a7776e4ffc744ca